### PR TITLE
Fix typo: lists -> lints

### PR DESCRIPTION
**Pull request by camelid**
It's a bit of a tongue-twister it seems.

### CODE REVIEW

File: src/bug-fix-procedure.md [hunk 1]
```diff
@@ -276,7 +276,8 @@ FutureIncompatibleInfo {
 
 Remove this too.
 
-#### Add the lint to the list of removed lists.
+<a name="add-the-lint-to-the-list-of-removed-lists">
```
Comment (camelid) [inline src/bug-fix-procedure.md:279]: I added an anchor so that the rename doesn't break existing links.

**Review by fmease — APPROVED** [review]

File: src/bug-fix-procedure.md (hunk 1 above)
Comment (fmease) [reply src/bug-fix-procedure.md:279]: Thanks, good idea! However, I'm not sure if it's actually needed, we break URL a lot for rustc-dev-guide

Comment (fmease) [reply src/bug-fix-procedure.md:279]: see ParamEnv PR and my HIR ty lowering PR which moved lowering to ast-lowering etc

Comment (camelid) [reply src/bug-fix-procedure.md:279]: > Thanks, good idea! However, I'm not sure if it's actually needed, we break URL a lot of times for rustc-dev-guide

Fair enough, I'll remove it. I was just thinking of a particular place in rustdoc that links here that I didn't want to break. But it's obvious enough what it's referring to anyway. :)

**Pull request merged**